object_store/src/azure/credential.rs

```diff
@@ -338,13 +342,15 @@ impl ClientSecretOAuthProvider {
 }
 
 #[async_trait::async_trait]
-impl TokenCredential for ClientSecretOAuthProvider {
+impl TokenProvider for ClientSecretOAuthProvider {
+    type Credential = AzureCredential;
+
     /// Fetch a token
     async fn fetch_token(
         &self,
         client: &Client,
         retry: &RetryConfig,
-    ) -> Result<TemporaryToken<String>> {
+    ) -> crate::Result<TemporaryToken<Arc<AzureCredential>>> {
```

Comment:
```suggestion
    ) -> crate::Result<TemporaryToken<Arc<Self::Credential>>> {
```

probably not worth the change, but I find this a little bit more explicit when reading the code.

Reply:
I think I prefer being explicit about the type returned